server/src/main/java/org/elasticsearch/health/HealthService.java

```diff
@@ -82,8 +93,25 @@ public List<HealthIndicatorResult> getHealth(@Nullable String indicatorName, boo
 
         Stream<HealthIndicatorResult> filteredIndicatorResults;
         if (clusterHealthIsObtainable) {
+            HealthInfo healthInfo;
+            if (HealthNode.isEnabled()) {
+                try {
+                    ActionFuture<FetchHealthInfoCacheAction.Response> responseActionFuture = client.execute(
+                        FetchHealthInfoCacheAction.INSTANCE,
+                        new FetchHealthInfoCacheAction.Request()
+                    );
+                    FetchHealthInfoCacheAction.Response response = responseActionFuture.actionGet();
```

Comment:
A few things here: 
- the "fetch health request" will not create a cancellable task (similar to https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/admin/indices/diskusage/AnalyzeDiskUsageShardRequest.java#L42 )
- we shouldn't be blocking the calling thread here - so let's use the async call (listener etc)
- we should add a timeout here as the response to the health API should be prompt (also the timeout will be used as an indication that we don't have health information)

Reply:
I've made the change for the 2nd. The timeout is set (well currently not set) in TransportHealthNodeAction. Does 10s sound reasonable for all health node actions? Or do we need to make this configurable differently for different actions? I would think 10s would be fine because all of this data is fairly time-sensitive and all of the actions ought to be very fast.

Reply:
These transport actions should be very quick. Let's go with a setting that defaults to `5s`